presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java

```diff
@@ -512,64 +504,71 @@ void reset()
             }
         }
 
-        private boolean test(long value)
+        int adjustValue(long value, int offsetIndex)
+        {
+            if (inDictionaryStream != null) {
+                if (!inDictionaryFlags[offsetIndex]) {
+                    value += stripeDictionarySize;
+                }
+            }
+            return toIntExact(value);
+        }
+
+        private boolean test(int dictionaryId)
```

Comment:
I haven't seen a reply to this comment:

---
I think dictionaryPosition is a better name because the value is a position in a dictionary block.